UIAutomationTests/TestFixture.cs

```diff
@@ -0,0 +1,207 @@
+﻿using FlaUI.Core;
+using FlaUI.Core.AutomationElements;
+using FlaUI.Core.Conditions;
+using FlaUI.Core.Input;
+using FlaUI.Core.WindowsAPI;
+using FlaUI.UIA3;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UIAutomationTests
+{
+    public class TestFixture : IDisposable
+    {
+
+        string vsExecutablePath = $@"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\devenv.exe";
+        string slnPath = $@"C:\repos\powerbi-exasol\Exasol\Exasol.mproj";
+
+        string queryPqPath = $@"C:\repos\powerbi-exasol\Exasol\Exasol.query.pq";
+        string originalQueryPqStr;
+
+        Application app;
+        UIA3Automation automation;
+
+        ConditionFactory cf;
+
+        AutomationElement mainWindow;
+
+        AutomationElement debugTargetButtonAE;
+        Button debugTargetButton;
+        AutomationElement MQueryOutput;
+        AutomationElement[] tabItemAEs;
+
+        //https://stackoverflow.com/questions/12976319/xunit-net-global-setup-teardown
+        //Do "global" initialization here; Only called once.
+        public TestFixture()
+        {
+            PrepVisualStudio();
+            SetupConditionFactory();
+            AcquireDebugTargetButton();
+            PressDebugTargetButton();
+            AcquireMQueryWindowAndAcquireTabsWhenFullyLoaded();
+            //entering the credentials seems to be more reliable than loading them
+            EnterCredentials();
+            SaveAndBackupOriginalQueryString();
+        }
+
+        private void SetupConditionFactory()
+        {
+            cf = new ConditionFactory(new UIA3PropertyLibrary());
+        }
+
+        private void EnterCredentials()
+        {
+            //the errors tab will pop up and ask for credentials
+            var errorsTabAE = tabItemAEs[2];
+            errorsTabAE.AsTabItem().Select();
+
+            var comboBoxes = WaitUntilMultipleFoundAsync(errorsTabAE, FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.ComboBox)).Result;
+            var cbCredentialType = comboBoxes[1].AsComboBox();
+            cbCredentialType.Select(0);
+
+            var textBoxes = errorsTabAE.FindAll(FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.Edit));
+            textBoxes[0].AsTextBox().Text = "sys";
+            textBoxes[1].AsTextBox().Text = "exasol";
+
+            var buttons = errorsTabAE.FindAll(FlaUI.Core.Definitions.TreeScope.Descendants, cf.ByControlType(FlaUI.Core.Definitions.ControlType.Button));
+            buttons[0].AsButton().Invoke();
+
```

Comment:
would be nice to extract every section to its own dedicated method with a proper name helpful to the user